### PR TITLE
Added intersection over union metric for bbox

### PR DESCRIPTION
**Pull request by Mainak431**
Bounding Box should be in the format x1,x2,y1,y2. x1,y1 left upper corner of the box in the image and x2,y2 be the right  lower corner of the box in the image. This metric is very useful as it can be used in many object localization tasks.

### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [x] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [x] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR!

Our current plan for such metrics is to maintain API compatibility with `tf.keras`. This means that we will be adding a version of the IoU metric in keras-team/keras, based on a Metric base class, like here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/metrics.py

@pavithrasv is responsible for this part of the API. If this is something you would like to be working on, I'm sure @pavithrasv will have development items that you can take on.

We cannot merge this PR in the current state since it would conflict with our API sync plans. I'll let @pavithrasv explain what needs to be done :)
